tests/conftest.py

```diff
@@ -228,134 +235,90 @@ def create_resource(kube_client, kube_ns, resource_type, manifest, delete_first=
-    return create_hub_pod(kube_client, kube_ns)
-
-
-@pytest.fixture
-def hub(hub_pod):
-    """Return the jupyterhub Hub object for passing to Spawner constructors
-
-    Ensures the hub_pod is running
-    """
-    return Hub(ip=hub_pod.status.pod_ip, port=8081)
-
-
-@pytest.fixture(scope="session")
-def hub_pod_ssl(kube_client, kube_ns, ssl_app):
-    """Start a hub pod with internal_ssl enabled"""
-    # load ssl dir to tarfile
-    buf = io.BytesIO()
-    tf = tarfile.TarFile(fileobj=buf, mode="w")
-    tf.add(ssl_app.internal_certs_location, arcname="internal-ssl", recursive=True)
-
-    # store tarfile in a secret
-    b64_certs = base64.b64encode(buf.getvalue()).decode("ascii")
-    secret_name = "hub-ssl-secret"
-    secret_manifest = V1Secret(
-        metadata={"name": secret_name}, data={"internal-ssl.tar": b64_certs}
-    )
-    create_resource(kube_client, kube_ns, "secret", secret_manifest)
-
-    name = "hub-ssl"
-
-    service_manifest = V1Service(
-        metadata=dict(name=name),
-        spec=V1ServiceSpec(
-            type="ClusterIP",
-            ports=[V1ServicePort(port=8081, target_port=8081)],
-            selector={"hub-name": name},
-        ),
-    )
-
-    create_resource(kube_client, kube_ns, "service", service_manifest)
-
-    return create_hub_pod(
-        kube_client,
-        kube_ns,
-        pod_name=name,
-        ssl=True,
-    )
-
-
-@pytest.fixture
-def hub_ssl(kube_ns, hub_pod_ssl):
-    """Return the Hub object for connecting to a running hub pod with internal_ssl enabled"""
-    return Hub(
-        proto="https",
-        ip=f"{hub_pod_ssl.metadata.name}.{kube_ns}",
-        port=8081,
-        base_url="/hub/",
-    )
+# Base create_moneypenny_pod on this et seq
+# def create_hub_pod(kube_client, kube_ns, pod_name="hub", ssl=False):
+#     config_map_name = pod_name + "-config"
+#     secret_name = pod_name + "-secret"
+#     with open(jupyterhub_config_py) as f:
+#         config = f.read()
+
+#     config_map_manifest = V1ConfigMap(
+#         metadata={"name": config_map_name},
+#         data={"jupyterhub_config.py": config},
+#     )
+
+#     config_map = create_resource(
+#         kube_client,
+#         kube_ns,
+#         "config_map",
+#         config_map_manifest,
+#         delete_first=True,
+#     )
+
+#     volumes = [{"name": "config", "configMap": {"name": config_map_name}}]
+#     volume_mounts = [
+#         {
+#             "mountPath": "/etc/jupyterhub/jupyterhub_config.py",
+#             "subPath": "jupyterhub_config.py",
+#             "name": "config",
+#         }
+#     ]
+#     if ssl:
+#         volumes.append(
+#             {"name": "secret", "secret": {"secretName": secret_name}}
+#         )
+#         volume_mounts.append(
+#             {
+#                 "mountPath": "/etc/jupyterhub/secret",
+#                 "name": "secret",
+#             }
+#         )
+
+#     pod_manifest = V1Pod(
+#         metadata={
```

Comment:
Yeah this test file has me confused?  This looks like a bunch of fixtures, which seem reasonable, but the mentions of hub and jupyter confuse me.  And I'm not sure which of these functions are the actual tests?  Can we move those to a different python file in this directory?

Reply:
This bit isn't done yet.  I'm stealing the harness from KubeSpawner for the actual things-that-do-things-in-k8s tests, and I'll have some tests around in-memory representations for K8s objects not submitted to kubectl.